test/unit/test_conf.ml

```diff
@@ -0,0 +1,41 @@
+let test_check_version =
+  let test name ~expected ~exe res =
```

Comment:
It's a bit surprising that `~expected` is not the expected test output here. What do you think about using `~config` to mean that it's the value found in the config file ("specification" might work as well)? That way `res` can be renamed to a more standard `expected`.

Actually this applies to `check_version` as well, it seems that `config:string -> exe:string -> ...` describes better what this function does.